Sources/Internal/LayoutItem.swift

```diff
@@ -66,13 +70,26 @@ extension LayoutItem.ItemType: Comparable {
       return lhsMonth < rhsMonth
     case let (.dayOfWeekInMonth(_, lhsMonth), .day(rhsDay)):
       return lhsMonth <= rhsDay.month
+    case let (.dayOfWeekInMonth(_, lhsMonth), .monthFooter(rhsMonth)):
+      return lhsMonth < rhsMonth
 
     case let (.day(lhsDay), .day(rhsDay)):
       return lhsDay < rhsDay
     case let (.day(lhsDay), .monthHeader(rhsMonth)):
       return lhsDay.month < rhsMonth
     case let (.day(lhsDay), .dayOfWeekInMonth(_, rhsMonth)):
       return lhsDay.month < rhsMonth
+    case let (.day(lhsDay), .monthFooter(rhsMonth)):
+      return lhsDay.month < rhsMonth
+
+    case let (.monthFooter(lhsMonth), .monthHeader(rhsMonth)):
+      return lhsMonth <= rhsMonth
+    case let (.monthFooter(lhsMonth), .dayOfWeekInMonth(_, rhsMonth)):
+      return lhsMonth < rhsMonth
+    case let (.monthFooter(lhsMonth), .day(rhsDay)):
+      return lhsMonth <= rhsDay.month
+    case let (.monthFooter(lhsMonth), .monthFooter(rhsMonth)):
+      return lhsMonth < rhsMonth
```

Comment:
```suggestion
    case let (.monthFooter(lhsMonth), .monthHeader(rhsMonth)):
      return lhsMonth < rhsMonth
    case let (.monthFooter(lhsMonth), .dayOfWeekInMonth(_, rhsMonth)):
      return lhsMonth < rhsMonth
    case let (.monthFooter(lhsMonth), .day(rhsDay)):
      return lhsMonth < rhsDay.month
    case let (.monthFooter(lhsMonth), .monthFooter(rhsMonth)):
      return lhsMonth < rhsMonth
```

Now the inverse fix - a footer is only considered "less than" another item in the calendar if that other item is in a month in the future. 

If the other item is in the _same month_ as the footer on the lefthand side of the comparison `<`, then there's no way that the footer can be "less than" that other item, since the footer is the last item in the month. In other words, the footer is the last item in each month, so it can't be less than any other item in that same month.